lib/dpl/provider/bintray.rb

```diff
@@ -0,0 +1,532 @@
+require 'json'
+require 'net/http'
+require "uri"
+require 'find'
+
+module DPL
+  class Provider
+    class Bintray < Provider
+
+      @file
+      @user
+      @key
+      @passphrase
+      @url
+      @descriptor
+      @dry_run
+      @test_mode = false
+
+      def check_auth
+      end
+
+      def needs_key?
+        false
+      end
+
+      def set_test_mode
+        @test_mode = true
+      end
+
+      def init_from_args
+        @user = options[:user]
+        @key = options[:key]
+        @url = options[:url]
+        @file = options[:file]
+        @passphrase = options[:passphrase]
+        @dry_run = options[:dry_run]
+
+        if @user.nil?
+          abort("The 'user' argument is required")
+        end
+        if @key.nil?
+          abort("The 'key' argument is required")
+        end
+        if @file.nil?
+          abort("The 'file' argument is required")
+        end
+        if @url.nil?
+          @url = 'https://api.bintray.com'
+        end
+        if @dry_run.nil?
+          @dry_run = false
+        end
+      end
+
+      def get_user
```

Comment:
In Ruby, I suggest not using `get_*` and `set_*` as method names.

Here, we can just write:

```ruby
attr_reader :user
```
and use `user` to get the value `@user`.